network/router_test.go

```diff
@@ -63,6 +63,52 @@ func testRouter(t *testing.T, fac routerFactory) {
 	}
 }
 
+// Test connection of multiple Hosts and sending messages back and forth
+// also tests for the counterIO interface that it works well
+func TestRouterErrorHandling(t *testing.T) {
+	h1, err1 := NewTestRouterTCP(2109)
+	h2, err2 := NewTestRouterTCP(2110)
+	if err1 != nil || err2 != nil {
+		t.Fatal("Could not setup hosts")
+	}
+
+	go h1.Start()
+	go h2.Start()
+
+	defer func() {
+		h1.Stop()
+	}()
+
+	// tests the setting error handler
+	require.Nil(t, h1.errorHandler)
+	errHandlerCalled := false
```

Comment:
Haha, I forgot the `-race`. I'm happy with cothority's Makefile, where `make test` does linting and testing with race etc...